src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -338,6 +343,18 @@ func (r *StatefulSetResource) obj(
 			fmt.Sprintf("--admin-api-tls-key %q", path.Join(resourcetypes.GetTLSMountPoints().AdminAPI.ClientCAMountDir, "tls.key")))
 	}
 
+	// In any case, configure PersistentVolumeClaimRetentionPolicy
+	// Default to old behavior: Retain PVC
+	// If auto-remove-pvcs flag is set, active new behavior: switch to Delete for both WhenScaled and WhenDeleted.
+	var pvcReclaimRetentionPolicy appsv1.StatefulSetPersistentVolumeClaimRetentionPolicy
+	if r.autoDeletePVCs {
+		pvcReclaimRetentionPolicy.WhenScaled = appsv1.DeletePersistentVolumeClaimRetentionPolicyType
+		pvcReclaimRetentionPolicy.WhenDeleted = appsv1.DeletePersistentVolumeClaimRetentionPolicyType
```

Comment:
The paranoia in me says we may want to keep when deleted as retain? I can't really imagine someone scaling the cluster manually but I could see cases where the STS needs to be deleted with orphan = true to work around it's immutability.

Reply:
Oh you're so right. GOOD CALL!
Delete on orphan DOES happen, and is actively used by the operator. Will change When deleted to RETAIN.

Reply:
actually need to test if orphan delete triggers this behavior..

Reply:
it looks like deleting with orphan does not get PVCs deleted.  obviously, since pods are retained as well. i can add a test. then it's fine to keep using delete on delete, what do you think?

Reply:
switched to Patch call. moved the patch helper func from cluster_controller to pkg/patch (pkg/utils does not work because of import conflict w/ types package)